soc/st/stm32/stm32h7x/soc_m7.c

```diff
@@ -49,6 +50,29 @@ static int stm32h7_m4_wakeup(void)
 }
 #endif /* CONFIG_STM32H7_DUAL_CORE */
 
+void soc_reset_hook(void)
+{
+	/* If the device tree has a chosen ITCM or DTCM node with
+	 * status okay, we need to initialize the values in the ECC
+	 * controller. To do so, we write all the locations with an arbitrary
+	 * value (0 in this case). This will prevent ECC error when
+	 * reading memory locations that have not been written yet.
+	 * This procedure is suggested in ST's AN5342 - How to use error
+	 * correction code (ECC) management for internal memories protection on
+	 * STM32 MCUs, section 3.1.1, Step 1
+	 */
+#if DT_NODE_HAS_STATUS_OKAY(DT_CHOSEN(zephyr_itcm)) && \
+	!DT_SAME_NODE(DT_CHOSEN(zephyr_itcm), DT_CHOSEN(zephyr_flash))
+	z_early_memset((void *)DT_REG_ADDR(DT_CHOSEN(zephyr_itcm)), 0,
+		       DT_REG_SIZE(DT_CHOSEN(zephyr_itcm)));
```

Comment:
Nit: earlier in the file, add something like
```c
#define STM32_ITCM DT_CHOSEN(zephyr_itcm)
```

which simplifies my request:
```suggestion
	uint64_t *itcm_start = (void*)DT_REG_ADDR(STM32_ITCM);
	uint64_t *itcm_end = (void*)((char*)itcm_start + DT_REG_SIZE(STM32_ITCM));
	for (uint64_t *p = itcm_start; p < itcm_end; p++) {
	    *p = 0;
	}
```

Reply:
```c
BUILD_ASSERT((ITCM_BASE & __alignof__(uint64_t)) == 0, "ITCM start address must be 64-bit aligned");
BUILD_ASSERT((ITCM_END & __alignof__(uint64_t)) == 0, "ITCM size must be 64-bit aligned");
```